uplink/src/collector/systemstats.rs

```diff
@@ -0,0 +1,546 @@
+use flume::Sender;
+use log::error;
+use serde::Serialize;
+use sysinfo::{DiskExt, NetworkData, NetworkExt, ProcessExt, ProcessorExt, System, SystemExt};
+use tokio::time::Instant;
+
+use std::{
+    collections::HashMap,
+    sync::Arc,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+use crate::base::{self, Buffer, Config, Package, Point, Stream};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Io error {0}")]
+    Io(#[from] std::io::Error),
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+pub struct SystemStat {
```

Comment:
For naming consistency:
`SystemStat` -> `System`
`Networks` -> `NetworkStats/NetworkStream`
`Disks` -> `DiskStats/DiskStream`
`SystemStats` -> `SystemStream` (if you've decided to go with `NetworkStream` instead of `NetworkStats`